app/models/article.rb

```diff
@@ -153,9 +158,11 @@ def self.unique_url_error
   validates :video_source_url, url: { allow_blank: true, schemes: ["https"] }
   validates :video_state, inclusion: { in: %w[PROGRESSING COMPLETED] }, allow_nil: true
   validates :video_thumbnail_url, url: { allow_blank: true, schemes: %w[https http] }
+  validate :future_or_current_published_at, on: :create
+  validate :has_correct_published_at?, on: :update, unless: :admin_update
```

Comment:
I don't like this solution: validation in a model and a hack to be able to updates as an admin. I would like to move validation logic to the places where we update or create articles (maybe put some logic in the `Articles::Attributes`, or reorganize it, I'll need to think more about it), I plan to work on this refactoring as one of my next tasks.

Reply:
I agree with your thoughts.  This implementation gets us the feature, but refactoring validation out of the article would be a tremendous improvement to our code.

Reading notes: adding model level validations over time often means that objects saved under previous validation constraints may no longer be valid.